tests/integration/frameworks/drupal/test_module_invoke_all.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*DESCRIPTION
+Test that a simple request through drupal_http_request gets instrumented.
+*/
+
+// force the framework to avoid requiring the drupal detection file
+/*INI
+newrelic.framework = drupal
+newrelic.application_logging.enabled = false
+newrelic.application_logging.forwarding.enabled = false
+newrelic.application_logging.metrics.enabled = false
+*/
+
+/*SKIPIF
+<?php
+require("skipif.inc");
+*/
+
+/*EXPECT
+g
+f
+f
+*/
+
+/*EXPECT_METRICS
+[
+  "?? agent run id",
+  "?? timeframe start",
+  "?? timeframe stop",
+  [
+    [{"name":"DurationByCaller/Unknown/Unknown/Unknown/Unknown/all"},
+                                                                    [1, "??", "??", "??", "??", "??"]],
+    [{"name":"DurationByCaller/Unknown/Unknown/Unknown/Unknown/allOther"},
+                                                                    [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/Logging/Forwarding/PHP/disabled"},    [1, "??", "??", "??", "??", "??"]],
+    [{"name": "Supportability/Logging/Metrics/PHP/disabled"},       [1, "??", "??", "??", "??", "??"]],
+    [{"name":"Framework/Drupal/Hook/f"},                [1, "??", "??", "??", "??", "??"]],
+    [{"name":"Framework/Drupal/Hook/g"},                [1, "??", "??", "??", "??", "??"]],
+    [{"name":"Framework/Drupal/Module/module"},         [2, "??", "??", "??", "??", "??"]],
+    [{"name":"OtherTransaction/all"},                   [1, "??", "??", "??", "??", "??"]],
+    [{"name":"OtherTransaction/php__FILE__"},           [1, "??", "??", "??", "??", "??"]],
+    [{"name":"OtherTransactionTotalTime"},              [1, "??", "??", "??", "??", "??"]],
+    [{"name":"OtherTransactionTotalTime/php__FILE__"},  [1, "??", "??", "??", "??", "??"]],
+    [{"name":"Supportability/framework/Drupal/forced"}, [1,    0,    0,    0,    0,    0]]
+  ]
+]
+*/
+
+require_once(realpath(dirname(__FILE__)) . '/../../../include/config.php');
+
+function module_invoke_all($f) {
+    call_user_func_array("module_" . $f, array());
+}
+
+function module_f() {
+    echo "f\n";
+}
+
+function module_g() {
+    echo "g\n";
+    module_f();
+}
+
+module_invoke_all("g");
+module_invoke_all("f");
```

Comment:
Maybe add a test for a hook with arguments? Something like:
```
function module_hook_with_arg($arg) {
    echo "module_hook_with_arg(arg=[$arg])\n";
}
module_invoke_all("hook_with_arg", "arg_value");
```
This would require a slight modification of `module_invoke_all` mock to actually pass the arguments. This should do the trick:
```
function module_invoke_all($f) {
  $args = func_get_args();
  unset($args[0]);
  call_user_func_array("module_" . $f, $args);
}
```
Additionally consider testing negative cases: 
* `module_invoke_all` is called with non-existing function
* `module_invoke_all` is called with a function expecting arguments but didn't receive any
* `module_invoke_all` is called with empty argument
I'm not sure these scenarios deserve tests in this PR, but would be good to know how the agent will handle these scenarios. Maybe these cases belong to #574 or to a subsequent PR?

Reply:
Added the first part in [d6412ef](https://github.com/newrelic/newrelic-php-agent/pull/559/commits/d6412ef0cfa8c82bfa93a8a263f691071ed741f6)

The three negative cases cannot really test agent functionality because each one throws a fatal PHP error.